omnibus/files/private-chef-ctl-commands/cleanup_bifrost.rb

```diff
@@ -0,0 +1,213 @@
+require 'pg'
+
+CREATE_SQL =<<SQL
+BEGIN;
+CREATE TABLE IF NOT EXISTS cleanup_tracking_auth_actors(
+    authz_id  CHAR(32)
+);
+
+CREATE OR REPLACE RULE cleanup_auth_actor_creation_tracking AS ON INSERT TO auth_actor DO INSERT INTO cleanup_tracking_auth_actors VALUES (NEW.authz_id);
+COMMIT;
+SQL
+
+DELETE_SQL =<<SQL
+BEGIN;
+DROP RULE IF EXISTS cleanup_auth_actor_creation_tracking ON auth_actor;
+DROP TABLE IF EXISTS cleanup_tracking_auth_actors;
+COMMIT;
+SQL
+
+add_command_under_category "cleanup-bifrost", "cleanup", "Cleanup orphaned bifrost objects.", 2 do
+  cleanup_args = ARGV[3..-1]
+  options = {}
+
+  OptionParser.new do |opts|
+    opts.banner = "chef-server-ctl cleanup-bifrost [options]"
+    opts.on("-b SIZE", "--batch-size", "How many authz actors to delete at a time") do |b|
+      options[:batch_size] = b.to_i
+    end
+
+    opts.on("--force-cleanup", "Clean up tracking tables (does not scan bifrost actors)") do |b|
+      options[:cleanup_only] = true
+    end
+
+    opts.on("--estimate-only", "Print estimate of orphaned authz objects") do |b|
+      options[:estimate_only] = true
+    end
+  end.parse!(cleanup_args)
+
+  if !options[:batch_size]
+    options[:batch_size] = 10000
+  elsif options[:batch_size] <= 0
+    puts "Invalid batch size: #{options[:batch_size]}"
+    exit(1)
+  end
+
+  if options[:cleanup_only]
+    remove_bifrost_tracking_table(bifrost_db)
+    exit(0)
+  end
+
+  if options[:estimate_only]
+    print_and_return_estimate(known_actor_list, bifrost_db)
+    exit(0)
+  end
+
+  run_cleanup(bifrost_db, options[:batch_size])
+end
+
+def run_cleanup(bifrost_db, batch_size)
+  safety_check(bifrost_db)
+  install_bifrost_tracking_table(bifrost_db)
+  # NOTE(ssd) 2017-07-24:
+  # For the most part, the tracking table manages in-flight requests for us.
+  # However, it is still possible that the following sequence happens:
+  #    - new authz_id in bifrost database
+  #    - tracking table inserted
+  #    - known_actor_list fetched
+  #    - new record in oc-chef database
+  #
+  # In the this case, the authz_id won't be in our list of known
+  # clients, won't be in our tracking table, but will show up in our
+  # search for orphaned authz_ids.
+  #
+  # Thus, this sleep is a 1-time wait to avoid that scenario.
+  #
+  puts "One-time sleep to account for in-flight requests not captured by tracking table"
+  sleep 25
+  begin
+    estimated_deletion_count = print_and_return_estimate(known_actor_list, bifrost_db)
+
+    if estimated_deletion_count <= 0
+      puts "Estimated deletion count 0. Aborting"
```

Comment:
I think this line will follow [that output](https://github.com/chef/chef-server/pull/1339/files#diff-3b35eccc49cc7c77455e83f80f22ff8eR129):
```
 Total chef users+clients: 100
Total bifrost auth_actors: 100
Deletion Candidates (est): 0
```
So it should be understandable from context, I think?